src/file_pipe_log/pipe.rs

```diff
@@ -322,9 +322,13 @@ impl<F: FileSystem> SinglePipe<F> {
         let mut writable_file = self.writable_file.lock();
         if writable_file.writer.offset() >= self.target_file_size {
             if let Err(e) = self.rotate_imp(&mut writable_file) {
+                // As the disk is already full and is no valid and extra directory to flush data, it can safely return the `no space err` to users.
+                if is_no_space_err(&e) && self.paths.len() == 1 {
```

Comment:
```suggestion
                // As the disk is already full and is no valid directory to flush data, it can safely return
                // the `no space err` to users.
                if is_no_space_err(&e) {
```

Reply:
As mentioned, no needs to check whether `paths.len() == 1`.